element-connector/src/main/java/org/eclipse/californium/elements/tcp/TcpClientConnector.java

```diff
@@ -59,10 +67,11 @@ public TcpClientConnector(int numberOfThreads, int connectTimeoutMillis, int idl
 		this.numberOfThreads = numberOfThreads;
 		this.connectionIdleTimeoutSeconds = idleTimeout;
 		this.connectTimeoutMillis = connectTimeoutMillis;
-		this.listenUri = URI.create(String.format("%s://127.0.0.1:0", getSupportedScheme()));
+		this.listenUri = URI.create(String.format("%s://0.0.0.0:0", getSupportedScheme()));
```

Comment:
replace the "localhost" with "any". "localhost" is usually only for loopback and not reachable for other host.